src/js/modules/core/components/MastheadBar.js

```diff
@@ -67,36 +72,62 @@ const navButtonStyles = {
     fontSize: '12px',
     fontWeight: 'bold',
     textTransform: 'uppercase',
+  },
+  barsIcon: {
+    marginRight: '6px',
+  },
+  searchIcon: {
+    marginRight: '3px',
   }
 };
 
-const NavButton = ({ classes, label }) => {
+const NavButton = ({ classes, label, onClick }) => {
+  const createIcon = () => {
```

Comment:
I know that. The whole point is that NavButton doesn't need to know what it's containing. By using a switch statement you're coupling what the NavButton contains with the NavButton. Now, sometimes you'll want to do this. But now, I think a child component makes the most sense